lib/streamlit/elements/widgets/multiselect.py

```diff
@@ -347,6 +347,8 @@ def _multiselect(
             multiselect_proto.value[:] = serde.serialize(widget_state.value)
             multiselect_proto.set_value = True
 
+        if ctx:
```

Comment:
since I am not too familiar with the testing framework yet: does this only resolve to `True` in the testing-scenario? If yes, it might make sense to leave a comment or rename `ctx` to make it more obvious

Reply:
No. It is true any time you are running streamlit. The only time the ScriptRunContext is not available is if you are running a streamlit script directly via python (`python my_app.py`)

Reply:
Thanks! So this means that `save_for_app_testing` is executed also in non-testing flows, right? Will `ctx.session_state[TESTING_KEY][k]` always raise a key-error that is caught in non-testing executions then for every re-run? If yes, do we expect any performance issues because of it? My first reaction is that we would want to check for something like `if IS_TEST_MODE:` or similar

Reply:
It will be executed in all flows, and raise the error in non-testing executions. I don't expect it to be much of a performance cost, as exceptions are pretty similar to regular `if` branches in terms of perf in python.

Reply:
I think I would also feel slightly better if we could introduce some kind of global flag or config option to indicate that it is in app testing mode. We have something very similar for the development mode as well:

<img width="527" alt="image" src="https://github.com/streamlit/streamlit/assets/103002573/b533414c-782b-46eb-ac68-574c3b23b7f7">

Having this flag could also be useful if we need to integrate other testing-related logic into Streamlit as well.

Reply:
I think I would also feel slightly better if we could introduce some kind of global flag or config option to indicate that it is in app testing mode. We have something very similar for the development mode as well:

<img width="527" alt="image" src="https://github.com/streamlit/streamlit/assets/103002573/b533414c-782b-46eb-ac68-574c3b23b7f7">

Having this flag could also be useful if we need to integrate other testing-related logic into Streamlit as well.

Reply:
Added a config option